include/deal.II/numerics/fe_field_function.templates.h

```diff
@@ -307,11 +312,17 @@ namespace Functions
       {
         fe_v.reinit(cells[i], i, 0);
         const unsigned int nq = qpoints[i].size();
-        std::vector< std::vector<Tensor<1,dim> > >
-        vgrads (nq, std::vector<Tensor<1,dim> >(n_components));
+        std::vector< std::vector<Tensor<1,dim,typename VECTOR::value_type> > >
+        vgrads (nq, std::vector<Tensor<1,dim,typename VECTOR::value_type> >(n_components));
         fe_v.get_present_fe_values ().get_function_gradients(data_vector, vgrads);
         for (unsigned int q=0; q<nq; ++q)
-          values[maps[i][q]] = vgrads[q];
+          {
+            std::vector< Tensor<1,dim> > &viq = values[maps[i][q]];
+            const unsigned int s = vgrads[q].size();
+            viq.resize(s);
+            for (unsigned int l=0; l<s; l++)
+              viq[l] = vgrads[q][l];
```

Comment:
I find this hard to read (with the writable reference `viq`. Can you avoid the temporary variable and just assign to `vgrads[maps[i][q]][l]`?